pkg/mcs/utils/util.go

```diff
@@ -70,6 +78,58 @@ func InitClusterID(ctx context.Context, client *clientv3.Client) (id uint64, err
 	return 0, errors.Errorf("failed to init cluster ID after retrying %d times", maxRetryTimes)
 }
 
+// GetExpectedPrimary indicates API has changed the primary.
+func GetExpectedPrimary(client *clientv3.Client, leaderPath string) string {
+	primary, err := etcdutil.GetValue(client, strings.Join([]string{leaderPath, ExpectedPrimary}, "/"))
+	if err != nil {
+		log.Error("get expected primary key error", errs.ZapError(err))
+		return ""
+	}
+
+	return string(primary)
+}
+
+// RemoveExpectedPrimary removes the expected primary key.
+// - removed when campaign new primary successfully.
+// - removed when appoint new primary by API.
+func RemoveExpectedPrimary(client *clientv3.Client, leaderPath string) {
+	log.Info("remove expected primary key", zap.String("leader-path", leaderPath))
+	// remove expected leader key
+	resp, err := kv.NewSlowLogTxn(client).
+		Then(clientv3.OpDelete(strings.Join([]string{leaderPath, ExpectedPrimary}, "/"))).
+		Commit()
+	if err != nil || !resp.Succeeded {
+		log.Error("change expected primary error", errs.ZapError(err))
+		return
+	}
+}
+
+// SetExpectedPrimary sets the expected primary key when the current primary has exited.
+func SetExpectedPrimary(client *clientv3.Client, leaderPath string) {
```

Comment:
The expected primary may be confused, How about something like xxxflag or something else?

Reply:
changed, PTAL thx!